minor tweak

developer9998 opened the pull request:
-updated credits tab
-possibly fixed names and colours not changing for others

hopefully i didnt fuck this up

developer9998 committed:
minor tweak

-updated credits tab
-possibly fixed names and colours not changing for others

hopefully i didnt fuck this up

developer9998 committed:
fixed name and colour stuff

yeahhhh :///

Graicc commented on ComputerInterface/BaseGameInterface.cs:
```diff
@@ -58,15 +60,15 @@ public static void SetName(string name)
-                bool leftHanded = GorillaComputer.instance != null ? GorillaComputer.instance.leftHanded : true;
-                GorillaTagger.Instance.myVRRig.photonView.RPC("InitializeNoobMaterial", RpcTarget.All, color.r, color.g, color.b, leftHanded);
-            }
+                GorillaTagger.Instance.myVRRig.photonView.RPC("InitializeNoobMaterial", RpcTarget.All, color.r, color.g, color.b, GorillaComputer.instance == null || GorillaComputer.instance.leftHanded);
```
```suggestion
                GorillaTagger.Instance.myVRRig.photonView.RPC("InitializeNoobMaterial", RpcTarget.All, color.r, color.g, color.b, GorillaComputer.instance?.leftHanded ?? true);
```

Graicc commented on ComputerInterface/Views/GameSettings/NameSettingView.cs:
```diff
@@ -31,6 +32,20 @@ private void Redraw()
             str.AppendLine();
             str.BeginColor("ffffff50").Append("> ").EndColor().Append(_textInputHandler.Text).AppendClr("_", "ffffff50");
 
+            /* dunno if you would want this
+            if (!_textInputHandler.Text.IsNullOrWhiteSpace() && !GorillaNetworking.GorillaComputer.instance.CheckAutoBanListForName(_textInputHandler.Text))
+            {
+                str.BeginColor("7F0000").AppendLines(2);
+                str.Append("* This name was blacklisted by the name filter.").EndColor();
+            }
+
```
This just leads to people trying things until they manage to circumvent the name filter. In a perfect world, CI would also ban on bad names, but I don't want to risk false bans. 
```suggestion
```

developer9998 replied:
didn't think about that

Graicc commented on ComputerInterface/Views/GameSettings/NameSettingView.cs:
```diff
@@ -31,6 +32,20 @@ private void Redraw()
+                str.Append("* This name was blacklisted by the name filter.").EndColor();
+            }
+
+            if (!_textInputHandler.Text.IsNullOrWhiteSpace() && _textInputHandler.Text.Length > 12)
+            {
+                str.BeginColor("7F0000").AppendLines(2);
+                str.Append("* This name will not be seen on unmodded devices.").EndColor();
+            }
+            */
+
```
CI really should just cap the name length to prevent people from sending excess data, which could potentially cause issues with the anti-cheat

```suggestion
```

developer9998 replied:
from what ive heard people are getting banned from entering long roomnames with CI but thats unrelated

Graicc commented on ComputerInterface/Views/GameSettings/ColorSettingView.cs:
```diff
@@ -35,10 +35,10 @@ public override void OnShow(object[] args)
         {
             var str = new StringBuilder();
-            str.BeginColor(_color).Repeat("=", SCREEN_WIDTH).EndColor().AppendLine();
+            str.BeginColor("ffffff").Repeat("=", SCREEN_WIDTH).EndColor().AppendLine();
```
Assuming you made this change because `_color` was causing null exceptions? This suggestion would fix it if that's the case.
```suggestion
            str.BeginColor(_color ?? Color.white).Repeat("=", SCREEN_WIDTH).EndColor().AppendLine();
```

developer9998 replied:
not sure if i missed an update but the colour view i built with the source code and the colour view in the release looked different so i must've rushed to finish it or whatever

Graicc commented on ComputerInterface/Views/GameSettings/ColorSettingView.cs:
```diff
@@ -35,10 +35,10 @@ public override void OnShow(object[] args)
             str.BeginCenter().Append("Change Color").AppendLine();
             str.Append("Values are from 0 - 255").EndAlign().AppendLine();
-            str.BeginColor(_color).Repeat("=", SCREEN_WIDTH).EndColor().AppendLine();
+            str.BeginColor("ffffff").Repeat("=", SCREEN_WIDTH).EndColor().AppendLines(2);
```
See above

```suggestion
            str.BeginColor(_color ?? Color.white).Repeat("=", SCREEN_WIDTH).EndColor().AppendLines(2);
```

Graicc commented on ComputerInterface/BaseGameInterface.cs:
```diff
@@ -58,15 +60,15 @@ public static void SetName(string name)
         public static string GetName()
         {
-            return PhotonNetwork.LocalPlayer.NickName;
+            if (GorillaComputer.instance.savedName.IsNullOrWhiteSpace())
+                return PhotonNetwork.LocalPlayer.NickName;
+            else
+                return GorillaComputer.instance.savedName;
```
```suggestion
            return !GorillaComputer.instance.savedName.IsNullOrWhiteSpace() ? GorillaComputer.instance.savedName : GorillaComputer.instance.savedName;
```

Graicc committed:
Simplify checks

Graicc committed:
Simplify null checks

developer9998 committed:
fix for newest patch

this should also prevent this issue from happening again

developer9998 committed:
adjusted the keyboard

made the game keys move when you press them just like the ones that the mod adds. probably the worst way to do it but it's still something.

developer9998 committed:
pesky bee

developer9998 committed:
Update ComputerInterface/Views/GameSettings/NameSettingView.cs

Co-authored-by: Graic <33105645+Graicc@users.noreply.github.com>

developer9998 committed:
Update ComputerInterface/Views/GameSettings/NameSettingView.cs

Co-authored-by: Graic <33105645+Graicc@users.noreply.github.com>

developer9998 committed:
Update ComputerInterface/Views/GameSettings/ColorSettingView.cs

Co-authored-by: Graic <33105645+Graicc@users.noreply.github.com>

developer9998 committed:
Update ComputerInterface/Views/GameSettings/ColorSettingView.cs

Co-authored-by: Graic <33105645+Graicc@users.noreply.github.com>
